addons/loyalty/models/loyalty_reward.py

```diff
@@ -230,6 +230,8 @@ def write(self, vals):
             # Keep the name of our discount product up to date
             for reward in self:
                 reward.discount_line_product_id.write({'name': reward.description})
+        if 'active' in vals:
+            self.reward_product_id.toggle_active()
```

Comment:
So... If we archive a reward... The linked reward product gets archived too?
And if we unarchived a reward whose product isn't archived, then it gets archived? Did I get that right? 
And what about the other products from the `reward_product_ids`? Even if they are just computed, shouldn't they have the same logic?
In the end, I'm pretty sure that's not something we want to do... '^^ 
I don't even think we want to magically reactivate an archived product when unarchiving the reward... For me, these are two different things, and this at least need a confirmation box, or just a direct Error explaining the product in question is archived, and the reward can't be unarchived as long as the product is archived.

Reply:
I don't really agree, as we are discussing about reward product such as "10% on next SO", not a storable product like a chair. So I don't think it is a big issue to (un-)archive straight away both together.
But you're right, I need to change the `toggle_active` as I was assuming that they had the same active value, but that might not be the case.